README.md

```diff
@@ -0,0 +1,175 @@
+# UWFE Firmware
+
+This is the firmware monorepo for the UW Formula Electric team. This repository contains vehicle board firmware, HIL board firmware, and our firmware validation testbed.
+
+# Onboarding Instructions
+
+- [OpenProject Access](#open-project)
+- [SSH Key](#ssh-key-set-up)
+- [Vagrant (Windows Only)](#Vagrant-Environment-Set-Up-for-Windows-Users)
+- [Mac OS](#mac-os-set-up)
+- [Linux](#linux-set-up)
+- [Other Resources](#other-resources)
+
+# Open Project
+
+We use Open Project for task management. (todo: add instructions on how to get access lol)
+
+# SSH Key Set Up
+
+Please see [this page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account) for instructions. Feel free to message in #firmware on Slack with any questions!
+
+# Vagrant Environment Set Up for Windows Users
+
+### Overview
+
+Vagrant is an open-source software product for building and maintaining portable virtual software development environments. We will be using it to create an Ubuntu VM for VirtualBox.
+The repository we use is found [here](https://github.com/UWaterloo-Formula-Electric/vagrant).
+
+### Prerequisites
+
+- Install the latest version of [Vagrant](https://developer.hashicorp.com/vagrant/downloads)
+- Install the latest version of [VirtualBox](https://www.virtualbox.org/wiki/Downloads)
+- Install the matching [VirtualBox Extension Pack](https://www.virtualbox.org/wiki/Downloads) to get USB data
+  After installing the above, verify the Vagrant installation worked by opening a new command prompt or console, and checking that **vagrant** is available.
+
+```
+$ vagrant
+Usage: vagrant [options] <command> [<args>]
+
+    -v, --version                    Print the version and exit.
+    -h, --help                       Print this help.
+
+# ...
+```
+
+### Getting Started
+
+Make sure everything under **Prerequisites** has been installed, and that you have an SSH key added for your host computer to have GitHub access (check the link above).
+
+Then, run the following commands:
+
+```
+git clone git@github.com:UWaterloo-Formula-Electric/vagrant.git
+cd vagrant
+vagrant up
```

Comment:
Can we reduce information overload by removing info that is not super relevant? I have never used any Vagrant commands and I think most people just use VirtualBox to interact with Vagrant. What does everyone else think?

Reply:
Generally if we can keep the onboarding short and sweet, newcomers won't find it as confusing and the impression of the firmware team will be better

Reply:
Interesting, because I've only ever used the vagrant commands 😅

Reply:
lol okay maybe keep them in

Reply:
The virtual box way uses the GUI which is easier IMO but either works

Reply:
I suggest keeping it. I have only used the vagrant commands and would like this option to be available